icaruscode/Decode/DecoderTools/TriggerDecoderV3_tool.cc

```diff
@@ -793,24 +797,38 @@ namespace daq
     //
     // fill sbn::ExtraTriggerInfo::cryostats
     //
-    auto setCryoInfo
-      = [this,&extra=*fTriggerExtra,isFirstEvent=(triggerID <= 1)]
-        (std::size_t cryo, icarus::KeyValuesData const& data)
+    auto setCryoInfo = [
+      this,&extra=*fTriggerExtra,isFirstEvent=(triggerID <= 1),data=parsedData
+      ]
+        (std::size_t cryo)
       {
-        std::string const SIDE = (cryo == sbn::ExtraTriggerInfo::EastCryostat)
+        std::string const Side
+          = (cryo == sbn::ExtraTriggerInfo::EastCryostat) ? "EAST": "WEST";
+        std::string const CrSide = (cryo == sbn::ExtraTriggerInfo::EastCryostat)
           ? "Cryo1 EAST": "Cryo2 WEST";
+        // trigger logic: 0x01=adders; 0x02=majority; 0x07=both
+        std::string const triggerLogicKey = "MJ_Adder Source " + Side;
+        int const triggerLogicCode = data.hasItem(triggerLogicKey)
+          ? data.getItem(triggerLogicKey).getNumber<int>(0): 0;
+        sbn::bits::triggerLogicMask triggerLogicMask;
```

Comment:
Just to make sure I understand the additions: There is a new word (or two) in the trigger data packet string than before based on the adders (this one and another)? However, because nothing is being operated on in the online, this does not affect the data fragment creation and just necessitates updating the decoder.

Reply:
That is correct. Since there may be two triggers in the two cryostats at the same time, there are two new words, one per cryostat (`MJ_Adder Source EAST` and `MJ_Adder Source WEST`). If I remember the fragment creation code correctly, it will split the trigger string in words and read the one it is interested in: we can add freely as long as the required words are still present and not renamed.

Reply:
So this is partly true, it does this if we want it to. Otherwise the entire string as it is sent is preserved and sent along with the information in the event. The decoder then presumably has utilities to extract the information from either there directly or the different functions within the fragment.